backend/src/llm.py

```
@@ -0,0 +1,119 @@
+) -> list[AnnotationSchema]:
+    """Call a multimodal LLM via LiteLLM to detect UI elements."""
+    
+    model_config: dict[str, Any] = config.models.get(model_name, {})
+    model: str = model_config.get("model_code", model_name)
+    api_key: str | None = model_config.get("api_key")
+    base_url: str | None = model_config.get("base_url")
+
+    # Encode image in base64 for data URI transport
+    encoded_image = base64.b64encode(image_data).decode()
+    # Extract mime type (e.g., "image/png" -> "png")
+    image_extension = image_type.split('/')[-1] if '/' in image_type else image_type
```

Ruff (0.12.2)

Local variable `image_extension` is assigned to but never used

Remove assignment to unused variable `image_extension`

(F841)